Latest edits with agents registration to Sunfish

christian-pinto opened the pull request.

cayton reviewed (changes requested):
Very valuable additional content, thanks!
A couple things:

-globally
- What is the policy on the capitalization of 'Agent' and 'Agents' and 'Managing Agent'?  We need to agree and use it the way we agree throughout the document.
- we need to search/replace incorrect capitalization of 'redfish' - I see at least one time it is all lowercase in the document which means we probably have it other places as well.

- 4.1.4.1
- Can we use another word than 'overlooks'?  Perhaps 'oversees'?
- need to correct the wording "...in the for of a RedFish tree."
- need to fix the "atthe" word.
- need a space between `ConnectionMethod` and 'field' ("`ConnectionMethod`field")

- 4.3.1
- Remove the 'all' from "... before all its resources..." - should just be "... before its resources..."

christian-pinto commented:
@cayton 
> * What is the policy on the capitalization of 'Agent' and 'Agents' and 'Managing Agent'?  We need to agree and use it the way we agree throughout the document.

I believe we could keep it lower case unless in a title because of camel casing.

> * we need to search/replace incorrect capitalization of 'redfish' - I see at least one time it is all lowercase in the document which means we probably have it other places as well.

I have replaced all instances with `Redfish` as it is the way it's used in the website.

> * 4.1.4.1
> * Can we use another word than 'overlooks'?  Perhaps 'oversees'?
> * need to correct the wording "...in the for of a RedFish tree."
> * need to fix the "atthe" word.
> * need a space between `ConnectionMethod` and 'field' ("`ConnectionMethod`field")
> * 4.3.1
> * Remove the 'all' from "... before all its resources..." - should just be "... before its resources..."

Fixed all the above

I'll push the new version once we agree.

@rherrell would you mind having a look at this PR, thanks?

rherrell reviewed (approved):
Looks good.
Two suggestions:
+481 -- Should we be clear here that the "...UUID assigned by Sunfish during the registration process" ?


+572 -- Given the example lists the new resource as the /redfish/v1/Fabrics/CXL branch, the description on this line is ambiguous about the breadth of retrieval of resources.  The reader might interpret "....result of the crawling process is for the advertised resource, and all child resources, ...." to mean *just* the Fabrics/CXL branch is brought into Sunfish.   Sunfish will bring in all resources linked to a resource under the Fabrics/CXL branch even they are in the Systems or Chassis branches.

Make the above changes if you agree.... otherwise ignore them.

christian-pinto commented:
Thanks @rherrell 

> Looks good. Two suggestions: +481 -- Should we be clear here that the "...UUID assigned by Sunfish during the registration process" ?

I have added a reference to the section where the agent registration is discussed and we describe that sunfish generates a 129-bits UUID associated with the agent.

> +572 -- Given the example lists the new resource as the /redfish/v1/Fabrics/CXL branch, the description on this line is ambiguous about the breadth of retrieval of resources. The reader might interpret "....result of the crawling process is for the advertised resource, and all child resources, ...." to mean _just_ the Fabrics/CXL branch is brought into Sunfish. Sunfish will bring in all resources linked to a resource under the Fabrics/CXL branch even they are in the Systems or Chassis branches.

Good point here. I have reworded to make sure we state that the entire sub-tree + linked resources are brought into the Sunfish tree.

christian-pinto commented:
Adde all the edits discussed above. @cayton can you please review the changes and finalize the review so that I can merge it ?

Thanks.
